src/pages/InputDemo/InputDemo.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react';
+import * as yup from 'yup';
+import TextField from '../../components/TextField/TextFeild';
+import SelectField from '../../components/SelectField/SelectField';
+import RadioGroup from '../../components/RadioGroup/RadioGroup';
+import Button from '../../components/button/button';
+import { radioOptionsFootball, selectOptions, radioOptionsCricket } from '../../config/constants';
+
+class InputDemo extends Component {
+  constructor(props) {
+    super(props);
+    this.schema = yup.object().shape({
+      name: yup.string().required('Please enter your name').min(3, 'Please enter no less than 3 characters'),
+      sport: yup.string().required('Please select a sport'),
+      cricket: yup.string().when('sport', {
+        is: 'cricket',
+        then: yup.string().required('What you do is required'),
+      }),
+      football: yup.string().when('sport', {
+        is: 'football',
+        then: yup.string().required('What you do is required'),
+      }),
+    });
+
+    this.state = {
+      name: '',
+      sport: '',
+      cricket: '',
+      football: '',
+      touched: {
+        name: false,
+        sport: false,
+        cricket: false,
+        football: false,
+      },
+    };
+  }
+
+  onChangeTextField = (event) => {
+    // console.log('Inside onChangeTextField ', event.target.value);
+    this.setState({ name: event.target.value });
+  }
+
+  onChangeSelectOptions = (e) => {
+    let { sport, cricket, football } = this.state;
+    sport = e.target.value;
+    if(sport === 'select')
+      sport='';
+    cricket = '';
+    football = '';
+    this.setState({ sport, cricket, football }, () => {
+
+    });
+  }
+
+  onChangeRadioOption = (e) => {
+    const { sport } = this.state;
+    let { cricket, football } = this.state;
```

Comment:
why have you intialized the variables with state values if you are not using them just assigning with new values